File: 1261. Find Elements in a Contaminated Binary Tree/fault.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @description https://leetcode.com/problems/find-elements-in-a-contaminated-binary-tree/
 * @type self
 * @param {TreeNode} root
 */
var FindElements = function(root) {
  this.root = root;
  this.root.val = 0;

  function traverse(node) {
    if (!node) {
      return;
    }
    if (node.left) {
      node.left.val = 2 * node.val + 1;
    }
    if (node.right) {
      node.right.val = 2 * node.val + 2;
    }
    traverse(node.left);
    traverse(node.right);
  }

  traverse(this.root);
};

/** 
 * @param {number} target
 * @return {boolean}
 */
FindElements.prototype.find = function(target) {
  return preorder(this.root).includes(target);
};

/** 
 * Your FindElements object will be instantiated and called as such:
 * var obj = new FindElements(root)
 * var param_1 = obj.find(target)
 */

function preorder(root) {
  if (!root) {
    return [];
  }
  return [root.val].concat(preorder(root.left), preorder(root.right));
}
/**
 * 28 / 31 test cases passee
 * this solution is time limit exceeded
 */

File: 55. Jump Game/fault.java
/**
 * https://leetcode.com/problems/jump-game/
 * self
 */
class Solution {
  public boolean canJump(int[] nums) {
    if (nums.length == 1) return true;
    int[] dp = new int[nums.length - 1];
    for (int i = 0; i < dp.length; i++) {
      dp[i] = nums.length - 1 - i;
    }
    if (nums[0] >= dp[0]) return true;
    for (int i = 1; i < dp.length; i++) {
      if (nums[i] >= dp[i] && nums[i - 1] > 0) return true;
    }
    return false;
  }
}